src/schroedinger.jl

```diff
@@ -102,3 +104,18 @@ function check_schroedinger(psi::Bra, H)
     check_multiplicable(psi, H)
     check_samebases(H)
 end
+
+
+_promote_time_and_state(tspan, psi0, f) = _promote_time_and_state(tspan, psi0, f(first(tspan), psi0))
+function _promote_time_and_state(tspan, psi0, H::AbstractOperator)
+    # general case is Ts<:Complex, Tt<:Real
+    Ts = eltype(H)
+    Tt = real(Ts)
+    (isconcretetype(Ts) && isconcretetype(Tt)) || @warn "For using `ForwardDiff` on `schroedinger` the element type of `real(H(t,psi)*psi)` must be concrete !!\nGot elements of type $Tt \nTry promoting the Hamiltonian elements based on the parameters you are differentiating by."
```

Comment:
In what situations would this warning message be seen? I feel it might be confusing as it will show up in front of users that have not touched `ForwardDiff` or `schroedinger`.

Reply:
I'll remove it.
This was trying to warn about something like
```
ω0, α = 5.0, -0.2 # not promoting
#...
fun = [fω, fα, fΩ]
H_at_t = LazySum([f(0.0) for f=fun], op)
@show H_at_t.factors #  type is abstract !! Complex{Real}
```
in the definition of `get_Ht`.
But It doesn't cover all possible "bad" definitions of `get_Ht`.